YAFC/Utils/CommandLineParser.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System.IO;
+using System;
+
+namespace YAFC {
+    public static class CommandLineParser {
+        public static string lastError { get; private set; } = string.Empty;
+        public static bool helpRequested { get; private set; }
+
+        public static bool errorOccured => !string.IsNullOrEmpty(lastError);
+
+        public static ProjectDefinition Parse(string[] args) {
+            ProjectDefinition options = new ProjectDefinition();
+
+            if (args.Length == 0) {
+                return options;
+            }
+
+            if (!args[0].StartsWith("--")) {
+                options.dataPath = args[0];
+                if (!Directory.Exists(options.dataPath)) {
+                    lastError = $"Data path '{options.dataPath}' does not exist.";
+                    return null;
+                }
+            }
+
+            for (int i = string.IsNullOrEmpty(options.dataPath) ? 0 : 1; i < args.Length; i++) {
+                switch (args[i]) {
+                    case "--mods-path":
+                        if (i + 1 < args.Length && !IsKnownParameter(args[i + 1])) {
+                            options.modsPath = args[++i];
+
+                            if (!Directory.Exists(options.modsPath)) {
+                                lastError = $"Mods path '{options.modsPath}' does not exist.";
+                                return null;
+                            }
+                        }
+                        else {
+                            lastError = "Missing argument for --mods-path.";
+                            return null;
+                        }
+                        break;
+
+                    case "--project-file":
+                        if (i + 1 < args.Length && !IsKnownParameter(args[i + 1])) {
+                            options.path = args[++i];
+
+                            if (!File.Exists(options.path)) {
+                                lastError = $"Project file '{options.path}' does not exist.";
+                                return null;
+                            }
+                        }
+                        else {
+                            lastError = "Missing argument for --project-file.";
+                            return null;
+                        }
+                        break;
+
+                    case "--expensive":
+                        options.expensive = true;
+                        break;
+
+                    case "--help":
+                        helpRequested = true;
+                        break;
+
+                    default:
+                        lastError = $"Unknown argument '{args[i]}'.";
+                        return null;
+                }
+            }
+
+            return options;
+        }
+
+        public static void PrintHelp() {
+            Console.WriteLine("Usage:");
```

Comment:
That multiline print can be improved by using [verbatim](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/strings/#verbatim-string-literals) string literals. It allows to write multiline outputs without `WriteLine` for each line.
```
Console.WriteLine("Usage:
YAFC");

/* Output: 
Usage:
YAFC
*/
```

Reply:
Update: fixed the comment above -- forgot @.

Reply:
Done.